src/RootIO.jl

```diff
@@ -57,10 +204,23 @@ function Fill(tree::TTree, data)
         end
     else
         println("Filling the TTtree with a row")
-        row = map(field -> getfield(data, field), fieldnames(typeof(data)))
+        row = data
+        if !(typeof(row) <: Vector)
+            row = map(field -> getfield(data, field), fieldnames(typeof(data)))
+        end
         for i in 1:length(tree._branch_array)
             if typeof(row[i]) <: CxxWrap.StdVector
                 ROOT.SetObject(tree._branch_array[i], row[i])
+            elseif typeof(row[i]) == String
+                str = row[i]
+                GC.@preserve str begin
+                    ROOT.SetBranchAddress(tree._ROOT_ttree, string(tree._branch_names[i]), convert(Ptr{Int8}, pointer(str)))
+                end
+            elseif typeof(row[i]) == Bool
+                b = fill(row[i])
+                GC.@preserve b begin
+                    ROOT.SetBranchAddress(tree._ROOT_ttree, string(tree._branch_names[i]), convert(Ptr{Int8}, pointer(b)))
+                end
             else
                 ROOT.SetAddress(tree._branch_array[i], Ref(row[i]))
```

Comment:
You need to keep an instance of Ref(row[i]) to then preserve it from GC.

You have an issue in L229 below. What needs to be preserved is the data variables, because they are used by the backend code of `Fill()`, while they are not referenced in the arguments. See the `read_tree3.jl` example of `ROOT.jl`

Reply:
I read more about Julia GC. I've preserved the entire row and kwargs(in the above function). Please look into it if it's correct now